cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -75,16 +82,31 @@ def update_nodes(strategy, nodelist)
     end
   end
 
-  def get_queues_with_changes
+  def get_all_queues(config)
+    # Get all queue names from the cluster config
+    slurm_queues = config.dig("Scheduling", "SlurmQueues")
+    queues_name = Set.new
+    slurm_queues.each do |queue|
+      queues_name.add(queue["Name"])
+    end
+    queues_name
+  end
+
+  def get_queues_with_changes(config)
     # Load change set and find queue with changes
     queues = Set.new
     change_set = JSON.load_file("#{node['cluster']['shared_dir']}/change-set.json")
     Chef::Log.debug("Loaded change set (#{change_set})")
-    change_set["changeSet"].each do |change|
-      next unless change["updatePolicy"] == "QUEUE_UPDATE_STRATEGY"
-      queue = change["parameter"][/Scheduling\.SlurmQueues\[([^\]]*)\]/, 1]
-      Chef::Log.info("Adding queue (#{queue}) to list of queue to be updated")
-      queues.add(queue)
+    if are_mount_or_unmount_required? # Changes with SHARED_STORAGE_UPDATE_POLICY require all queues to update
+      queues = get_all_queues(config)
+      Chef::Log.info("All queues will be updated in order to add new shared storage")
```

Comment:
In the log we mention storage addition only ("add new shared storage"), but we should refer also to the removal or more generally to an update, right?

Reply:
This logic can only be touched when using queue update strategy "DRAIN" or "TERMINATE". Removal of storages will require compute fleet stop, the logs will not show up. I changed it to `All queues will be updated in order to update shared storage`.  Does it work for you?

Reply:
It works